### PR TITLE
Added Timestamp and Date Datatype conversion For Snowflake

### PR DESCRIPTION
**Pull request by Jithendar12**
Issue#INC00001071:Data types conversion (TIMESTAMP_NTZ, DATE, TIMESTAMP --> TIMESTAMPMILLI) is not working.
These columns are getting converted to VARCHAR

Description of changes:
Previsouly TIMESTAMP and DATE Datetypes are casted into TIMESTAMPMILLI which converting those DataTypes to VARCHAR On Athena. Now we have casted TIMESTAMP to DATEMILLI and DATE to DATEDAY which converts into appropriate DataType on Athena.

Please refer attached document for more details and sample queries.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
[snowflake_datatype_issue.docx](https://github.com/awslabs/aws-athena-query-federation/files/9044412/snowflake_datatype_issue.docx)

### CODE REVIEW

File: athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java [hunk 1]
```diff
@@ -365,19 +365,26 @@ public Schema getSchema(Connection jdbcConnection, TableName tableName, Schema p
...
                     ) {
-                        columnType = Types.MinorType.TIMESTAMPMILLI.getType();
+                        columnType = Types.MinorType.DATEMILLI.getType();
+                    }
+                    if (dataType != null && (dataType.equalsIgnoreCase("TIMESTAMP_LTZ"))
+                    ) {
+                        columnType = Types.MinorType.DATEMILLI.getType();
                     }
                     if (dataType != null && (dataType.equalsIgnoreCase("TIMESTAMP_NTZ"))
                     ) {
```
Comment (henrymai) [inline athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java:382]: Can all of these if statements just be replaced with a static final class level Map (even higher up before my selection here), like:

    static final Map<String, ArrowType> stringToArrowTypeMap = Map.of(
        entry("DECIMAL", Types.MinorType.BIGINT.getType()),
        entry("INTEGER", Types.MinorType.INT.getType()),
        // ....
     );

And then in this code you can just do: `ArrowType columnType = stringToArrowTypeMap(dataType)`
Deleting the original definition of columnType where it just gets overwritten anyway.

Comment (Jithendar12) [reply athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java:382]: Sure @henrymai, will update with this

Comment (henrymai) [reply athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java:382]: Also I just noticed something, the original code is mapping "DECIMAL" to Arrow BIGINT, is this intentional?
Usually "DECIMAL" types should be mapped to Arrow DECIMAL instead.

Comment (henrymai) [reply athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java:382]: Looking at the Snowflake documentation confirms this too.

DECIMAL is the same as NUMBER and NUMBER is definitely not just a BIGINT:
https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#number

Additionally the code doesn't parse the parameters to DECIMAL (or NUMBER) which specify the precision and scale.

Note that these comments aren't suggesting that you fix this in this pull request, its just something that I noticed and is a potential bug that should be fixed at some point.

Comment (Jithendar12) [reply athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java]: HI Henry,done changes as per your suggestions.please review

Comment (henrymai) [reply athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java]: Looked through it, I'm letting tests run on it now.

**Review by henrymai — APPROVED** [review]

**Pull request merged**